server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -352,6 +355,19 @@ private void snapshot(final ShardId shardId, final Snapshot snapshot, final Inde
         }
     }
 
+    /**
+     * Checks whether the index commit is from a sequence number that is equal to the shards global and local checkpoint.
+     *
+     * @param indexShard  shard
+     * @param indexCommit index commit
+     * @return true if max sequence number in the index commit is equal to the shard's global and local checkpoint
+     */
+    private static boolean isSafeIndexCommit(IndexShard indexShard, IndexCommit indexCommit) throws IOException {
+        final long localCheckPoint = indexShard.getLocalCheckpoint();
```

Comment:
We should use the local checkpoint from the commit (not the current local checkpoint) and the persisted global checkpoint.